### DISCUSSION TITLE
TI HDC3022 temperature/humidity sensor

### DISCUSSION THREAD

**Discussion by therobveiller** [arendst/Tasmota#22563] [Q&A]
Does anyone have experience of the TI HDC3022 sensor? I know this is not currently supported on Tasmota, and I'm not asking for it to be supported either, but this sensor claims to have a calibration accuracy of ±0.1 °C and ±0.5 %RH. I wonder if this is in fact valid in real-world testing by anyone out there? If it really is that good, I can buy one and use it on a supported device for use as a "good enough" calibartion source.

The HDC3022 itself is way too expensive to be used as a general purpose temp/humidity sensor with Tasmota IMHO, but I need something to accurately calibrate my go-to sensor, the HTU21. I have tried using a MCP9808 which claims a calibrated absolute max ±0.5 °C over an very wide range, with ±0.25 °C more typical, but my experience is that they are just as wildly uncalibrated and inaccurate as the HTU21 devices, and need TempOffset in Tasmota of several degrees C in some cases.

I have some catering probe thermometers that really do read 0 °C at freezing and 100 °C at boiling, so I have little doubt in their accuracy at normal room temperature ranges, and they all tally with cheap household room thermometers give or take 0.5 °C at those livable ranges. Far from ideal, but they are the best I can use for calibration at the moment - I need something genuinely better.

**Comment by sfromis** [comment]
Melting ice is a good calibration reference for temperature. And those body temperature thermometers are usually pretty close, within their range.

**Reply by therobveiller** [reply]
Quite agree sfromis, but the HDC3022 has an alledgedly accurate %RH too, unlike the HTU21 which differers hugely between modules. I don't want to have to mess around with wet blub/dry bulb calculations from 2 thermometers to get something supposedly more accurate.

**Reply by sfromis** [reply]
If you want a well-calibrated reference device, I'd suggest to avoid cheap IOT stuff. That specifically includes HDC3022 - which is not even expensive.

**Reply by therobveiller** [reply]
An HDC3022 costs £26 - I call that exspensive when an HTU21 and similar cost only £3 !

**Reply by sfromis** [reply]
No, it only costs about $4 for the chip (Digikey), and Adafruit has a nice breakout board for $9. Neither are bottom-price sources. That's far from being "expensive" sensors, and of course some other sensor types are even cheaper. Sure, if you do not look around, you may find someone wanting customers to pay more, maybe for being "local".

**Comment by arendst** [comment]
Or try the HYT939:

![image](https://github.com/user-attachments/assets/7122b87f-25de-4b90-a142-acf955ce0c3e)

**Reply by therobveiller** [reply]
Thanks Theo nice idea, but those are even more expensive (but allegedly accurate) - £59 just for the chip on Digikey, and crazy prices on eBay. Farnell UK have them at £30 each, but I must buy 100 or more - err, no thank you :-)
I can only assume they were readily available for sensible prices at one time, which is why you have one.

I'd still love to know of anyone's _experience_ of the HDC3022, regardless of its price if I shop around.
